src/components/Cart/BillingForm.tsx

```diff
@@ -0,0 +1,166 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Controller, useForm } from "react-hook-form";
+import { billingValidationShema } from "./validation";
+import FormField from "@components/FormComponents/FormField";
+import { Button } from "@components/Button/Button";
+
+import { BillingFormFields, ButtonVariant } from "../../main/types/enums";
+import { useEffect, useMemo } from "react";
+import { AppDispatch } from "../../redux/app/store";
+import { useDispatch } from "react-redux";
+import {
+  getCities,
+  getCountries,
+} from "../../redux/features/location/locationSlice";
+import { useAppSelector } from "../../redux/app/hooks";
+import {
+  transformCitiesToOptions,
+  transformCountriesToOptions,
+} from "../../main/helpers";
+import {
+  selectCities,
+  selectCountries,
+} from "../../redux/features/location/selectors";
+import { NotesField } from "./NotesField";
+import { LocationFields } from "./LocationFields";
+import { PolicyField } from "./PolicyField";
+
+interface BillingFormFieldProps {
+  name: BillingFormFields;
+  label: string;
+}
+
+export const BillingForm = () => {
+  const {
+    handleSubmit,
+    formState: { errors },
+    control,
+    watch,
+    resetField,
+  } = useForm({
+    resolver: yupResolver(billingValidationShema),
+    defaultValues: {
+      agreeToPolicy: false,
+      agreeToEmails: false,
+    },
+  });
+
+  const country = watch("country");
+  const dispatch: AppDispatch = useDispatch();
+  const countries = useAppSelector(selectCountries);
+  const cities = useAppSelector(selectCities);
+
+  useEffect(() => {
+    dispatch(getCountries());
+  }, []);
+
+  useEffect(() => {
+    if (country) {
+      dispatch(getCities(country));
+    } else {
+      resetField("city");
+    }
+  }, [country]);
+
+  const onSubmit = async () => {};
+
+  const countriesOptions = useMemo(
+    () => transformCountriesToOptions(countries),
+    [countries],
+  );
+
+  const citiesOptions = useMemo(
+    () => transformCitiesToOptions(cities),
+    [cities],
+  );
+
+  const BillingFormField = ({ name, label }: BillingFormFieldProps) => {
+    return (
+      <Controller
+        control={control}
+        name={name}
+        render={({ field, fieldState }) => (
+          <FormField
+            label={label}
+            placeholder={label}
+            {...field}
+            error={fieldState.error?.message}
+          />
+        )}
+      />
+    );
+  };
```

Comment:
It must be reusable input component on global level. Isn`t it?

Reply:
It is already changed